store/sqlstore/post_store.go

```diff
@@ -889,6 +890,39 @@ func (s *SqlPostStore) Search(teamId string, userId string, params *model.Search
 			searchQuery = strings.Replace(searchQuery, "POST_FILTER", "", 1)
 		}
 
+		// handle after: before: on: filters
+		if len(params.AfterDate) > 1 || len(params.BeforeDate) > 1 || len(params.OnDate) > 1 {
+			dateClause := ""
+
+			if len(params.OnDate) > 1 {
+				onDate := model.ParseDateFilterToTime(params.OnDate)
+
+				// between `on date` start of day and end of day
+				dateClause = fmt.Sprintf("AND CreateAt BETWEEN %d AND %d ", model.GetStartOfDayMillis(onDate), model.GetEndOfDayMillis(onDate))
```

Comment:
Instead of using `fmt.Sprintf` to add the dates, could you specify those using named parameters like `AND CreateAt BETWEEN :Before AND :After` and then setting them like `queryParams["Before"] = model.GetStartOfDayMillis(onDate)`? I don't think you could do any SQL injection using the results of `model.GetStartOfDayMillis`, but it'd be good to be safe and avoid it entirely

Reply:
no disagreement here

Reply:
@hmhealey updated this in the latest commit, take a look